src/lib.rs

```diff
@@ -456,16 +457,59 @@ impl SandboxFS {
     }
 }
 
+/// Creates a file `path` with the given `uid`/`gid` pair.
+///
+/// The file is created via the `create` lambda, which can create any type of file it wishes.  The
+/// `delete` lambda should match this creation and allow the deletion of the file, and this is used
+/// as a cleanup function when the ownership cannot be successfully changed.
+fn create_as<T, E, P, Create, Delete>(
+    path: &P, uid: unistd::Uid, gid: unistd::Gid, create: Create, delete: Delete) -> Result<T, E>
```

Comment:
I think this would be more readable if you used impl trait:

```rust
fn create_as<P: AsRef<Path>, T, E: From<Errno>>(path: &P, uid: unistd::Uid, gid: unistd::Gid, create: impl Fn(&P) -> Result<T, E>, delete: impl Fn(&P) -> Result<(), E>) -> Result<T, E>
```

(or use the where clause if you prefer it)

Reply:
It helps. Done.